### PR TITLE
Introduce `WpGmtDateTime`

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR introduces `WpGmtDateTime` which is a wrapper for [`chrono::DateTime<chrono::offset::Utc>`](https://docs.rs/chrono/latest/chrono/struct.DateTime.html).

We use `uniffi::custom_type!` macro to convert it into `java.util.Date` in Kotlin bindings using `i64` as the underlying value. To make sure this conversion is done correctly, we export a couple assertion functions in the native bindings. These assertion functions are used in Kotlin to make sure the dates are converted correctly between Rust and Kotlin. We'll do the same for Swift in a follow up PR.

### CODE REVIEW

File: wp_api/src/date.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct WpGmtDateTime(#[serde(with = "wp_utc_date_format")] pub DateTime<Utc>);
+
+impl WpGmtDateTime {
+    pub fn from_timestamp(seconds: i64) -> Self {
+        let date_time =
+            DateTime::<Utc>::from_timestamp(seconds, 0).unwrap_or(DateTime::<Utc>::UNIX_EPOCH);
+        Self(date_time)
+    }
```
Comment (crazytonyli) [inline wp_api/src/date.rs:14]: What do you think about adding a few unit tests for this function with different kinds of values (zero, positive, negative)?

**Review by crazytonyli — APPROVED** [review]

File: wp_api/src/date.rs (hunk 1 above)
Comment (oguzkocer) [reply wp_api/src/date.rs:14]: I don't mind it, but would it be useful? We'd be testing the underlying implementation for `DateTime::<Utc>` and I generally don't add those tests as I feel they don't add any value.

Btw, I am not too thrilled about the `.unwrap_or(DateTime::<Utc>::UNIX_EPOCH)` part, but with `custom_type!` I think we either do this or panic. Out of them, I'd prefer us to panic so that if there are any issues, we'd be aware of it. Otherwise, this may only show up as a support request. I kept your implementation from that temporary PR and I was thinking of bringing it up in the PR description.

**Comment by oguzkocer** [comment]
I am going to merge this as is so that we can follow it up with [the PR that adds Swift support.](https://github.com/Automattic/wordpress-rs/pull/551)

There is an open discussion about further unit tests and whether we should panic if we can't create a date from timestamp. If any further actions are necessary for these discussions, I'll open follow up PRs.

File: wp_api/src/date.rs (hunk 1 above)
Comment (crazytonyli) [reply wp_api/src/date.rs:14]: I thought ensuring we always use "second" as the unit might be useful. Plus, adding test for some distant past and future dates might be helpful to ensure `WpGmtDateTime` continues to support a large range of dates.

The only reason that the parsing can fail is that the date is super distant. I think the 1970 date is an okay fallback for this parsing error because the original super distant is probably a mistake anyway. At bare minimal, we should not panic when parsing them.

Comment (oguzkocer) [reply wp_api/src/date.rs:14]: Addressed in https://github.com/Automattic/wordpress-rs/pull/556.
